zkevm-circuits/src/rlp_circuit_fsm.rs

```diff
@@ -2262,6 +2220,196 @@ impl<F: Field> RlpCircuitConfig<F> {
+                meta.query_advice(byte_rev_idx, Rotation::cur()),
+            );
+            cb.require_equal(
+                "stack can only init once with the first byte",
+                meta.query_advice(byte_idx, Rotation::cur()),
+                1.expr(),
+            );
+            cb.require_zero(
+                "stack inits at depth 0",
+                meta.query_advice(rlp_decoding_table.depth, Rotation::cur()),
+            );
+            cb.require_zero(
+                "stack inits with al_idx at 0",
+                meta.query_advice(rlp_decoding_table.al_idx, Rotation::cur()),
+            );
+            cb.require_zero(
+                "stack inits with sk_idx at 0",
+                meta.query_advice(rlp_decoding_table.sk_idx, Rotation::cur()),
+            );
+            cb.condition(
+                not::expr(meta.query_fixed(q_first, Rotation::cur())),
+                |cb| {
+                    cb.require_boolean(
+                        "tx_id can only stay the same or increment by 1",
+                        meta.query_advice(rlp_decoding_table.tx_id, Rotation::cur())
+                            - meta.query_advice(rlp_decoding_table.tx_id, Rotation::prev()),
+                    );
+                },
+            );
+            cb.require_equal(
+                "Init Op has DecodeTagStart in the state machine",
+                is_decode_tag_start(meta),
+                1.expr(),
+            );
+
+            cb.gate(and::expr([
+                meta.query_fixed(q_enabled, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.is_stack_init, Rotation::cur()),
+            ]))
+        });
+
+        // Lookup strategies for PUSH:
+        // 1. The input expressions include the indicators for each depth level.
+        // 2. For each active depth indicator, it's checked against the allowable transitional
+        //    scenarios in the state machine for that depth.
+        // 3. The byte_idx and depth transition are included.
+        meta.lookup_any("Decoding table stack op PUSH correspondence", |meta| {
+            let enable = meta.query_advice(rlp_decoding_table.is_stack_push, Rotation::cur());
+
+            let input_exprs = vec![
+                meta.query_advice(rlp_decoding_table.tx_id, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.format, Rotation::cur()),
+                meta.query_advice(rlp_decoding_table.byte_idx, Rotation::cur()),
+                1.expr(), // PUSH op increases depth
+                // List scenario on each depth.
+                // Sufficiency is achieved by listing all possible depth.
+                meta.query_advice(is_stack_depth_one, Rotation::cur()),
+                meta.query_advice(is_stack_depth_two, Rotation::cur()),
+                meta.query_advice(is_stack_depth_three, Rotation::cur()),
+                meta.query_advice(is_stack_depth_four, Rotation::cur()),
```

Comment:
We need to make sure the number of bytes pushed into higher depth must match the one sm extracts from `BeginObject / BeginVector` header. For a tag = `BeginObject / BeginVector` that starts with byte in the interval `[0xc0, 0xf8)`, the table expr for value should be `byte_value - 0xc0`; for tag starts with byte that are larger than `0xf8`, the table expr for value should be `tag_value_acc`. Therefore it's more reasonable to do lookup from sm into RlpDecodingTable.

Reply:
This correspondence is added in the latest commits.